tokio-stream/src/wrappers/watch.rs

```diff
@@ -0,0 +1,50 @@
+use crate::Stream;
+use async_stream::stream;
+use std::pin::Pin;
+use tokio::sync::watch::Receiver;
+
+use std::fmt;
+use std::task::{Context, Poll};
+
+/// A wrapper around [`tokio::sync::watch::Receiver`] that implements [`Stream`].
+///
+/// [`tokio::sync::watch::Receiver`]: struct@tokio::sync::watch::Receiver
+/// [`Stream`]: trait@crate::Stream
+pub struct WatchStream<T> {
+    inner: Pin<Box<dyn Stream<Item = ()>>>,
+    _marker: std::marker::PhantomData<T>,
+}
```

Comment:
It seems like a problem to have the `Stream` return `()`. The user of the `Stream` would want the actual item, though I think we might have to clone it to do that?

Reply:
🤔  I thought users would mostly use this to see whether something got arrived? Or maybe it is nice to have it return the last arrived value as well.

Reply:
Or can I just`yield rx.borrow()` ? Instead of yield a ()? I am trying this and now having some difficulty to deal with the life-times.